tooling/v8-snapshot/src/setup/generate-metadata.ts

```diff
@@ -0,0 +1,71 @@
+import { writeFileSync } from 'fs'
+import { BundlerMetadata, generateBundlerMetadata } from '../generator/snapshot-generate-entry-via-dependencies'
+import debug from 'debug'
+import { ensureDirSync } from 'fs-extra'
+import path from 'path'
+
+const logInfo = debug('cypress:snapgen:info')
+const logDebug = debug('cypress:snapgen:debug')
+const logError = debug('cypress:snapgen:error')
+
+async function createMeta ({
+  appEntryFile,
+  nodeModulesOnly,
+  pathsMapper,
+  projectBaseDir,
+  snapshotEntryFile,
+}) {
+  return generateBundlerMetadata(projectBaseDir, snapshotEntryFile, {
+    entryFile: appEntryFile,
+    pathsMapper,
+    nodeModulesOnly,
+  })
+}
+
+/**
+ * Generates the snapshot meta file by following all dependencies of
+ * the `appEntryFile` via `v8-snapshot`.
+ * This file is then used by the snapshot doctor to determine how to process
+ * the modules found inside the snapshot entry.
+ *
+ * @param {Partial<import('../snapconfig').SnapshotConfig>} opts
+ * @return {Promise<import('v8-snapshot').BundlerMetadata>} metadata
+ */
+export async function generateMetadata ({
+  appEntryFile,
+  metaFile,
+  nodeModulesOnly,
+  pathsMapper,
+  projectBaseDir,
+  snapshotEntryFile,
+}: {
+  appEntryFile: string
+  metaFile: string
+  nodeModulesOnly: boolean
+  pathsMapper: (file: string) => string
+  projectBaseDir: string
+  snapshotEntryFile: string
+}): Promise<BundlerMetadata> {
```

Comment:
is `meta` a promise when it is returned? it doesn't seem like it if you're able to write it to a file.

Reply:
This is a promise because the function is `async`